dbschema/prompt-variant-response.esdl

```diff
@@ -0,0 +1,13 @@
+module default {
+  type PromptVariantResponse extending Resource, Mixin::Embedded, Mixin::Owned {
+    promptId: str;
+    multi link responses: VariantResponse {
```

Comment:
It would be nice to include the constraint that the variant can only be used once.
Idk how ATM. Maybe at least a todo here for now?

Reply:
I did a little bit of research and it seems like there is no way to do this in the schema, but rather people will do it at the application level or even do it before inserting to check the list for duplicates

Reply:
Yeah it could be done if the `VariantResponse` owned the relationship to the `PromptVariantResponse`. Practically it's the same, it would just change inserts a little.

```esdl
type PromptVariantResponse {
  responses := .<x[is VariantResponse];
}

type VariantResponse {
  required x: PromptVariantResponse;
  required variant: str;
  response: RichText;

  constraint exclusive on ((.x, .variant));
}
```

Reply:
Do this now or alongside the queries? I added a TODO for now, but not sure which path forward you'd prefer

Reply:
It's hard to say for certain without getting my head into the queries. Something may force us one way or the other.
But I want to encode everything we can into the schema. The less we have to do in app code, data wise, the better.
So yeah, I'd do this now. Thanks 🙏 
Also rename `x` to something better...I'm too tired to think what it might be. I had a real hard time naming these objects since they are so generic. If anything `pvr`.